asmcomp/CSEgen.ml

```diff
@@ -253,16 +253,16 @@ method private kill_loads n =
 (* Perform CSE on the given instruction [i] and its successors.
    [n] is the value numbering current at the beginning of [i]. *)
 
-method private cse n i =
+method private cse n i k =
   match i.desc with
   | Iend | Ireturn | Iop(Itailcall_ind _) | Iop(Itailcall_imm _)
   | Iexit _ | Iraise _ ->
-      i
+      k i
   | Iop (Imove | Ispill | Ireload) ->
       (* For moves, we associate the same value number to the result reg
          as to the argument reg. *)
       let n1 = set_move n i.arg.(0) i.res.(0) in
-      {i with next = self#cse n1 i.next}
+      self#cse n1 i.next (fun next -> k { i with next; })
```

Comment:
FWIW, stylistically the rest of the file doesn't terminate records with a `;`

Reply:
Maybe not, but we're never going to get absolute consistency everywhere without using ocamlformat.  I've found trailing semicolons to be a good style to reduce unnecessary diffs when refactoring (admittedly more noticeable when records are formatted with one field per line).